core/src/mindustry/world/blocks/logic/MessageBlock.java

```diff
@@ -123,12 +125,27 @@ public void buildConfiguration(Table table){
                             dialog.hide();
                         }
                     });
+                    dialog.closeOnBack(() -> {
```

Comment:
My problem with this is that someone might accidentally edit something in, and want to close the block without applying changes. Applying it regardless makes it impossible.